source/helpers/DataStructure.js

```diff
@@ -0,0 +1,165 @@
+        }
+
+        Object.entries(value).forEach(valueObject => {
+          const [childFieldId, childValue] = valueObject;
+          const listItem = other.inputs.find(obj => obj.key === childFieldId);
+          const { tags: listItemTags } = listItem;
+
+          answers.push(
+            createAnswerObject({
+              fieldId: `${id}.${childFieldId}`,
+              value: childValue,
+              parentId: id,
+              tags: listItemTags ?? [],
+            })
+          );
+        });
+        return;
+
+      case 'avatarList':
+        if (value === undefined) {
+          return;
+        }
+
+        Object.entries(value).forEach(valueObject => {
+          const [childFieldId, childValue] = valueObject;
+          answers.push(
+            createAnswerObject({
+              fieldId: `${id}.${childFieldId}`,
+              value: childValue,
+              parentId: id,
+              tags,
+            })
+          );
+        });
+        return;
+
+      case 'repeaterField':
+        if (value === undefined) {
+          return;
+        }
+
+        Object.entries(value).forEach(repeaterField => {
+          const [childFieldId, childItems] = repeaterField;
+
+          Object.entries(childItems).forEach(childItem => {
+            const [repeaterItemId, repeaterItemValue] = childItem;
+            const repeaterFieldItem = other.inputs.find(obj => obj.id === repeaterItemId);
+            const { tags: repeaterFieldItemTags } = repeaterFieldItem;
+
+            answers.push(
+              createAnswerObject({
+                fieldId: `${id}.${childFieldId}.${repeaterItemId}`,
+                value: repeaterItemValue,
+                parentId: id,
+                tags: repeaterFieldItemTags ?? [],
+              })
+            );
+          });
+        });
+
+        return;
+
+      default:
+        answers.push(
+          createAnswerObject({
+            fieldId: id,
+            value,
+            tags,
+          })
+        );
+    }
+  });
+
+  return answers;
+};
+
+/**
+ * Checks if string is a numeric value
+ * @param {string} str
+ */
+const isNumeric = str => {
+  if (typeof str !== 'string') return false; // we only process strings!
+  return !isNaN(str) && !isNaN(parseFloat(str));
+};
+
+/**
+ * Converts case answers array to object
+ * Credits to Jacob for the magical data mapping
+ * @param {array} caseData
+ */
+export const convertAnswerArrayToObject = caseData => {
+  const caseObject = {};
+
+  caseData.answers.forEach(element => {
+    const path = element.field.id.split('.');
+    path.reduce((prev, curr, i) => {
+      if (!prev) {
+        return undefined;
+      }
+      if (!prev[curr]) {
+        if (i === path.length - 1) {
+          prev[curr] = element.value;
+        } else if (isNumeric(path[i + 1])) {
+          prev[curr] = [];
+        } else {
+          prev[curr] = {};
+        }
+      }
+
+      return prev[curr];
+    }, caseObject);
+  });
```

Comment:
This is `slightly' hard to read and understand. The naming should be improved, and some explanatory comments would be good to... For starters, we can rename element --> answer, and curr --> pathPart.